rudder-web/src/main/scala/com/normation/rudder/web/model/DirectiveFieldEditors.scala

```diff
@@ -550,167 +554,8 @@ class CheckboxField(val id: String) extends DirectiveField {
-          case "blank" => blank = true ; keepCurrent = false;
-          case _ => blank = false; keepCurrent = false;
-      } }, ("style" -> "margin-right:10px") )
-
-      (radios(0),radios(1),radios(2))
-    }
-
-    val form =
-      "zone=value *" #> initialPass &
-      "zonechooseHash" #> ( (xml:NodeSeq) => if(algos.size < 1) Text(s"hash with) ${algos.head.prefix.toUpperCase}") else xml ) &
-      "name=hash" #> (if(algos.size == 1) {
-                       Text(algos.head.prefix.toUpperCase)
-                     } else {
-                       SHtml.selectObj(algos.map(a => (a, a.prefix.toUpperCase))
-                                      , Box(currentAlgo)
-                                      , { (a:HashAlgoConstraint) => currentAlgo = Some(a) }
-                       )
-                     } ) &
-      ".radioKeep" #> radioKeep &
-      ".radioChange" #> radioChange &
-      "name=password" #> S.formGroup(10) { //use formGroup because must be the last evaluated method
-                           //always "" for default value
-                           SHtml.password(
-                               currentValue
-                                // ".get" should be licit at that point, because is only when reading from LDAP
-                             , {s => parseClient(newInternalValue(keepCurrent, blank, toClient, s, currentAlgo.get)) }
-                           )
-                         } &
-      "zone=blank" #>  { (nodes:NodeSeq) =>
-                         (if(blankable) {
-                           (".radioBlank" #> { (nodes:NodeSeq) => radioBlank }).apply(nodes)
-                         } else {
-                           NodeSeq.Empty
-                         })
-      }
-
-
-    Full(form.apply(PasswordField.xml))
-
-  }
-}
-
-object PasswordField {
-  val xml =
-    <table>
-      <tr><td colspan="2"><div zone="currentValue">Current password<span zone="value">(SHA256 hash): ce94806f8020be351fe3e492808a23a5b929c28a</span></div></td></tr>
-      <tr>
-        <td>New password:</td><td><input class="radioKeep" type="radio" name="[choice]" value="[true]" checked="checked"/>Keep current value</td>
-      </tr>
-      <tr>
-        <td></td>
-        <td>
-          <input class="radioChange" type="radio" name="[choice]" value="[true]"/>
-          <input type="text" name="password"/>
-          <span zone="chooseHash">hash with:
-            <select name="hash">
-              <option>MD5</option>
-            </select>
-          </span>
-        </td>
-      </tr>
-      <tr>
-        <td></td>
-        <td>
-          <div zone="blank">
-            <input class="radioBlank" type="radio" name="[choice]" value="[true]"/>None
-          </div>
-        </td>
-      </tr>
-    </table>
-}
 
-class MasterPasswordField(val id: String, blankable:Boolean, algos:Seq[HashAlgoConstraint]) extends DirectiveField {
+class PasswordField(val id: String, algos:Seq[HashAlgoConstraint])  extends DirectiveField {
```

Comment:
We thought they were always blankable ... maybe your right 

Would be easy to display delete button only if blankable